x/storage/keeper/cross_app_permission.go

```diff
@@ -243,3 +316,45 @@ func (app *PermissionApp) getResourceOwner(ctx sdk.Context, policy *permtypes.Po
 	}
 	return resOwner, nil
 }
+
+func (app *PermissionApp) getResourceOwnerAndIdFromGRN(ctx sdk.Context, grn types2.GRN) (resOwner sdk.AccAddress, resID math.Uint, err error) {
```

Comment:
why copy so many code, please use the `getResourceOwnerAndIdFromGRN` in the permission.go

Reply:
reject, getResourceOwnerAndIdFromGRN in the permission.go is a private function belong to Keeper

Reply:
accept,fixed